specification/redisenterprise/resource-manager/Microsoft.Cache/preview/2024-09-01-preview/redisenterprise.json

```diff
@@ -0,0 +1,2712 @@
+              }
+            ]
+          }
+        },
+        "evictionPolicy": {
+          "type": "string",
+          "description": "Redis eviction policy - default is VolatileLRU",
+          "enum": [
+            "AllKeysLFU",
+            "AllKeysLRU",
+            "AllKeysRandom",
+            "VolatileLRU",
+            "VolatileLFU",
+            "VolatileTTL",
+            "VolatileRandom",
+            "NoEviction"
+          ],
+          "x-ms-enum": {
+            "name": "EvictionPolicy",
+            "modelAsString": true
+          }
+        },
+        "persistence": {
+          "$ref": "#/definitions/Persistence",
+          "description": "Persistence settings",
+          "x-ms-identifiers": []
+        },
+        "modules": {
+          "type": "array",
+          "description": "Optional set of redis modules to enable in this database - modules can only be added at creation time.",
+          "items": {
+            "$ref": "#/definitions/Module"
+          },
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ],
+          "x-ms-identifiers": [
+            "name"
+          ]
+        },
+        "geoReplication": {
+          "type": "object",
+          "description": "Optional set of properties to configure geo replication for this database.",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ],
+          "properties": {
+            "groupNickname": {
+              "type": "string",
+              "description": "Name for the group of linked database resources",
+              "x-ms-mutability": [
+                "read",
+                "create"
+              ]
+            },
+            "linkedDatabases": {
+              "type": "array",
+              "description": "List of database resources to link with this database",
+              "items": {
+                "$ref": "#/definitions/LinkedDatabase"
+              }
+            }
+          }
+        },
+        "redisVersion": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Version of Redis the database is running on, e.g. '6.0'"
+        },
+        "deferUpgrade": {
+          "type": "string",
+          "description": "Option to defer upgrade when newest version is released - default is NotDeferred. Learn more: https://aka.ms/redisversionupgrade",
+          "enum": [
+            "Deferred",
+            "NotDeferred"
+          ],
+          "x-ms-enum": {
+            "name": "DeferUpgradeSetting",
+            "modelAsString": true
+          }
+        },
+        "accessKeysAuthentication": {
+          "type": "string",
+          "description": "Access database using keys - default is enabled. This property can be Enabled/Disabled to allow or deny access with the current access keys. Can be updated even after database is created.",
```

Comment:
>enabled [](http://example.com/codeflow?start=66&length=7)

I feel that in a PUT context, perhaps default behavior when the property is not included should mean 'do not change the vlaue', instead of 'Enabled', to prevent accidentally re-enabling access keys.

This would probably be more consistent with behavior of old api versions also!?